File: java/doit_java/src/chap06/exercise/Q6/CardCompany.java
/*
Q6
카드 회사에서 카드를 발급할 때마다 카드 고유 번호를 부여해 줍니다. 06-3의 <학번 생성하기>
예제를 참고하여 카드가 생성될 때마다 카드 번호가 자동으로 증가할 수 있도록 카드 클래스를
만들고 생성해 보세요.
 */
package chap06.exercise.Q6;

public class CardCompany {
    private static CardCompany instance = new CardCompany();

    private CardCompany() {}

    public static CardCompany getInstance() {
        if (instance == null)
            instance = new CardCompany();
        return instance;
    }

    public Card createCard() {
        Card card = new Card();
        return card;
    }
}

File: python/python_oop_practice/p_oop_2/chap05/excercise/card/card_company.py
from card import Card

class CardCompany:  # 싱글톤 패턴으로 카드회사 클래스 선언
    _instance = None

    @classmethod
    def _get_instance(cls):
        return cls._instance

    @classmethod
    def get_instance(cls, *args, **kargs):
        cls._instance = cls(*args, **kargs)
        cls.instance = cls._get_instance
        return cls._instance

    def create_card(self):  # 카드 발급
        card = Card()
        return card
